allennlp/data/dataset_readers/dataset_utils/text2sql_utils.py

```diff
@@ -0,0 +1,137 @@
+        mapped to table names/abstract variables.
+    sql : ``List[str]``
+        The tokens in the SQL query which corresponds to the text.
+    text_variables : ``Dict[str, str]``
+        A dictionary of variables associated with the text, e.g. {"city_name0": "san fransisco"}
+    sql_variables : ``Dict[str, str]``
+        A dictionary of variables associated with the sql query.
+    """
+    text: List[str]
+    text_with_variables: List[str]
+    sql: List[str]
+    text_variables: Dict[str, str]
+    sql_variables: Dict[str, str]
+
+
+def get_tokens(sentence: List[str],
+               sentence_variables: Dict[str, str]) -> List[str]:
+    """
+    Replaces abstract variables in text with their concrete counterparts.
+    """
+    tokens = []
+    for token in sentence:
+        if token not in sentence_variables:
+            tokens.append(token)
+        else:
+            for word in sentence_variables[token].split():
+                tokens.append(word)
+    return tokens
+
+def clean_and_split_sql(sql: str) -> List[str]:
+    """
+    Cleans up and unifies a SQL query. This involves removing uncessary quotes
+    and spliting brackets which aren't formatted consistently in the data.
+    """
+    sql_tokens = []
+    for token in sql.strip().split():
+        token = token.replace('"', "").replace('"', "").replace("%", "")
+        if token.endswith("(") and len(token) > 1:
+            sql_tokens.append(token[:-1])
+            sql_tokens.append(token[-1])
+        else:
+            sql_tokens.append(token)
+    return sql_tokens
+
+
+def process_sql_data_blob(data: JsonDict,
+                          use_all_sql: bool = False,
+                          use_question_split: bool = False,
+                          cross_validation_split: int = None) -> Iterable[Tuple[str, SqlData]]:
+    # If we're splitting based on SQL queries,
+    # we assign whole splits of questions which
+    # have a similar SQL template to the same split.
+    dataset_split: str = data['query-split']
+
+    # TODO(Mark): currently this does not filter for duplicate _sentences_
+    # which have the same sql query. Really it should, because these instances
+    # are literally identical, so just magnify errors etc. However, doing this
+    # would make it really hard to compare to previous work. Sad times.
+    for sent_info in data['sentences']:
+        # Instead, if we're using the question split,
+        # we take the split according to the individual question.
+        if use_question_split:
+            dataset_split = sent_info['question-split']
+        # We are observing the split we're meant to use for cross-validation.
+        # set the dataset split to be test. NOTE: This was _incorrect_ in the
+        # original repo, causing test leakage in datasets which used cross-validation.
+        if cross_validation_split is not None:
+            if str(cross_validation_split) == str(dataset_split):
+                dataset_bucket = "test"
```

Comment:
`dataset_bucket` vs. `dataset_split` is confusing.  I think you're using `dataset_bucket` here to mean "what bucket does the current sentence go to?"  If so, can this just be `sentence_split`?  Maybe this wouldn't be as confusing if there were a docstring explaining what exactly the inputs are (including the important pieces of the json blob you get as input), and that's all this needs...